crates/core/component/dex/src/component/router/fill_route.rs

```diff
@@ -81,7 +85,7 @@ async fn fill_route_inner<S: StateWrite + Sized>(
     hops: &[asset::Id],
     spill_price: Option<U128x128>,
     ensure_progress: bool,
-) -> Result<(Value, Value)> {
+) -> Result<Vec<Vec<Value>>> {
```

Comment:
I think this should return a `SwapExecution`, since it already computes `input` and `output`.